In [36]:
import pandas as pd

# 从CSV文件中加载数据
data = pd.read_csv('数据.csv')

In [7]:
#处理数据

feature_names = data.columns
feature_names = feature_names[:-1]
feature_names

Index(['问题1', '问题2', '问题3', '问题4', '问题5', '问题6', '问题7', '问题8', '问题9', '问题10',
       '问题11', '问题12', '问题13', '问题14', '问题15', '问题16', '问题17', '问题18', '问题19',
       '问题20', '问题21', '问题22', '问题23', '问题24', '问题25', '问题26', '问题27', '问题28',
       '问题29', '问题30', '问题31'],
      dtype='object')

In [12]:
#信度分析： 信度分析通常包括以下几种方法：

#Cronbach's Alpha： 用于测量问卷内部一致性的常见方法。可以使用pingouin库进行计算。

import pingouin as pg

# 计算Cronbach's Alpha
alpha = pg.cronbach_alpha(data[feature_names])
print('Cronbach Alpha:', alpha)

Cronbach Alpha: (0.8823638930987753, array([0.861, 0.902]))


计算的Cronbach's Alpha为0.8823638930987753，同时还提供了置信区间为0.861到0.902。Cronbach's Alpha是一种用于测量问卷内部一致性的统计指标，其值通常介于0和1之间，值越高表示问卷的内部一致性越强。  
在这种情况下，Cronbach's Alpha的值接近0.88，这表明问卷的内部一致性相对较高。置信区间提供了对Cronbach's Alpha估计的不确定性的一种度量，它告诉我们在统计意义上，Cronbach's Alpha可能在0.861到0.902之间。

进行KMO（Kaiser-Meyer-Olkin）检验，你可以使用Python中的factor_analyzer库。这个库提供了一个方便的接口来执行因子分析以及一些相关的统计检验，包括KMO检验和Bartlett的球形度检验。

In [18]:
import pandas as pd
from factor_analyzer.factor_analyzer import calculate_kmo, calculate_bartlett_sphericity

In [46]:
df = data.drop('score', axis=1)

In [47]:
# 计算KMO值
kmo_per_variable, kmo_total = calculate_kmo(df)
print("KMO值:", kmo_total)

KMO值: 0.9645421979824679


In [48]:
# 进行Bartlett的球形度检验
chi_square_value, p_value = calculate_bartlett_sphericity(df)
print("Bartlett球形度检验:")
print("近似卡方:", chi_square_value)
print("df:", data.shape[1] * (data.shape[1] - 1) // 2)  # 计算自由度
print("P:", p_value)

Bartlett球形度检验:
近似卡方: 4801.552295767867
df: 496
P: 0.0
